src/components/map/layers/edit/EditLinkLayer.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { Component } from 'react';
+import * as L from 'leaflet';
+import { withLeaflet } from 'react-leaflet';
+import { matchPath } from 'react-router';
+import { inject, observer } from 'mobx-react';
+import { IReactionDisposer, reaction } from 'mobx';
+import navigator from '~/routing/navigator';
+import SubSites from '~/routing/subSites';
+import { INode, ILink } from '~/models';
+import { EditNetworkStore } from '~/stores/editNetworkStore';
+import NodeMarker from '../mapIcons/NodeMarker';
+import { LeafletContext } from '../../Map';
+
+interface IEditLinkLayerProps {
+    editNetworkStore?: EditNetworkStore;
+    leaflet: LeafletContext;
+}
+
+@inject('editNetworkStore')
+@observer
+class EditLinkLayer extends Component<IEditLinkLayerProps> {
+    private reactionDisposers: IReactionDisposer[] = [];
+    private editableLinks: L.Polyline[] = [];
+
+    componentDidMount() {
+        this.reactionDisposers.push(reaction(
+            () => this.props.editNetworkStore!.links,
+            () => {
+                const links = this.props.editNetworkStore!.links;
+                if (links.length === 0) {
+                    this.removeOldLinks();
+                }
+            },
+        ));
+    }
+
+    private removeOldLinks = () => {
+        const map = this.props.leaflet.map;
+        // Remove (possible) previously drawn links from map
+        this.editableLinks.forEach((editableLink) => {
+            map!.removeLayer(editableLink);
+        });
+        this.editableLinks = [];
+    }
+
+    componentWillUnmount() {
```

Comment:
I don't find any code style rules about this currently, but I assume we usually try to put lifecycle methods at the bottom.

Reply:
This seems to be this way in other places as well, so no need to update this if you don't want to.